src/main/java/alns/Evaluator.java

```diff
@@ -94,12 +93,85 @@ private static int findSailingDuration(int startTime, Installation fromInstallat
         return ArcGeneration.hourToDiscTimePoint(distance / averageMaxSpeed);
     }
 
+    public static boolean evaluateVisitOrder(Solution solution) {
+        List<List<Order>> orderSequences = solution.getOrderSequences();
+        List<List<Integer>> installationSequences = getInstallationSequences(orderSequences);
+
+        if (installationInMultipleSequences(installationSequences)) {
+            return false;
+        }
+        
+        List<List<Order>> copyOfOrderSequences = Helpers.deepCopyList(orderSequences);
+        if (wrongVisitOrder(copyOfOrderSequences)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static List<List<Integer>> getInstallationSequences(List<List<Order>> orderSequences) {
+        List<List<Integer>> installationSequences = new ArrayList<>();
+
+        for (int vesselNumber = 0; vesselNumber < orderSequences.size(); vesselNumber++) {
+            List<Order> orderSequence = orderSequences.get(vesselNumber);
+            List<Integer> installationSequence = new ArrayList<>();
+            for (Order order : orderSequence) {
+                int installationID = order.getInstallationId();
+                installationSequence.add(installationID);
+            }
+
+            installationSequences.add(installationSequence);
+        }
+
+        return installationSequences;
+    }
+
+    private static boolean installationInMultipleSequences(List<List<Integer>> installationSequences) {
+        while (installationSequences.size() > 1) {
```

Comment:
For å komme oss bort fra kopieringen og fjerningen kan vi heller si
`for (int i = 0; i < instSequences.size(); i++) {
     int j = i + 1;
     while (j < instSequences.size()) {
          do stuff
          j++;
     }
}`

Reply:
Av en eller annen grunn får jeg ikke til line breaks i kommentarene men du skjønner kanskje greia